spec/commands/courses/update_status_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+RSpec.describe Courses::UpdateStatus do
+  let!(:course) { create(:course, :draft) }
+  let(:result) { subject.call }
+
+  describe '#call' do
+    context 'when invalid type of argument is passed' do
+      subject { described_class.new(current_user, course, 'ready') }
+
+      let(:current_user) { build(:user, :teacher) }
+
+      it 'raises a type constraint error' do
+        expect { subject }.to raise_error(Dry::Types::ConstraintError)
+      end
+
+      it 'does not update provided course with provided status' do
+        result rescue nil
+        course.reload
+        expect(course.status).not_to eq('ready')
+      end
+    end
+
+    context 'when valid type of argument is passed' do
+      context 'when provided course does not exist' do
+        subject { described_class.new(current_user, course.id + 1, 'ready') }
+
+        let(:current_user) { build(:user, :teacher) }
+
+        it 'raises a record not found error' do
+          expect { result }.to raise_error(ActiveRecord::RecordNotFound)
+        end
+
+        it 'does not update provided course with provided status' do
+          result rescue nil
+          course.reload
+          expect(course.status).not_to eq('ready')
+        end
+      end
+
+      context 'when provided course exists' do
+        subject { described_class.new(current_user, course.id, 'ready') }
+
+        context 'when not authorized' do
+          let(:current_user) { build(:user, :admin) }
+
+          it 'raises a not authorized error' do
+            expect { result }.to raise_error(Pundit::NotAuthorizedError)
+          end
+
+          it 'does not update provided course with provided status' do
+            result rescue nil
+            course.reload
+            expect(course.status).not_to eq('ready')
+          end
+        end
+
+        context 'when authorized' do
+          let(:current_user) { create(:user, :teacher) }
+          let(:expected_return) do
+            JSON.generate(
+              {
+                id: course.id,
+                subject: course.subject,
+                description: course.description,
+                author_id: course.author_id,
+                status: 'ready',
+                price: course.price,
+                language: course.language
+              }
+            )
+          end
+
+          before { course.update!(author_id: current_user.id) }
+
+          it 'does not raise an error' do
+            expect { result }.not_to raise_error
+          end
+
+          it 'does updates provided course with provided status' do
+            result rescue nil
+            course.reload
+            expect(course.status).to eq('ready')
```

Comment:
```suggestion
            expect { result }.to change { course.status }.from('draft').to('ready')
```

Then you can replace all three lines with just this one. Also doing `result rescue nil` here doesn't do anything as you wont get any error thrown.

Reply:
fixed